src/dash-table/derived/paginator.ts

```diff
@@ -19,8 +23,15 @@ export function lastPage(data: Data, page_size: number) {
 
 function getBackEndPagination(
     page_current: number,
-    setProps: SetProps
+    setProps: SetProps,
+    max_page_count: number | undefined
 ): IPaginator {
+
+    // adjust for zero-indexing
+    if (max_page_count) {
+        max_page_count = Math.max(0, max_page_count - 1);
```

Comment:
Should be able to handle an unknown `max_page_count` as this is a valid scenario. If not provided, could be `NaN` or `null` and prevent the use of `loadLast` for example. `max_page_count` prop should make it clear that the `null` case with BE with make some assumptions / deprecate the page navigation UI.

Reply:
I'm doing that in the actual button: https://github.com/plotly/dash-table/blob/f1d90daeea6129b0dd8b8ac8816074175f8a3a4b/src/dash-table/components/ControlledTable/index.tsx#L932

I can move it into the paginator if that's better, though.

Reply:
Yes please 🙏